src/unstructured.rs

```diff
@@ -523,6 +558,93 @@ impl<'a> Unstructured<'a> {
             _marker: PhantomData,
         })
     }
+
+    /// Call the given function an arbitrary number of times.
+    ///
+    /// The function is given this `Unstructured` so that it can continue to
+    /// generate arbitrary data and structures.
+    ///
+    /// You may optionaly specify minimum and maximum bounds on the number of
+    /// times the function is called.
+    ///
+    /// You may break out of the loop early by returning `Ok(false)`. You will
+    /// continue the loop by returning `Ok(true)`.
```

Comment:
There is a stable [std::ops::ControlFlow](https://doc.rust-lang.org/stable/std/ops/enum.ControlFlow.html) which seems much more appropriate here.